packages/create-amplify/src/initial_project_file_generator.ts

```diff
@@ -14,79 +13,37 @@ export class InitialProjectFileGenerator {
    */
   constructor(
     private readonly projectRoot: string,
-    private readonly fs = _fs,
-    private readonly executeWithDebugLogger = _executeWithDebugLogger
+    private readonly packageManagerController: PackageManagerControllerType,
+    private readonly fsp = _fsp,
+    private readonly path = _path
   ) {}
 
-  private readonly executableName = !process.env.PACKAGE_MANAGER_EXECUTABLE
-    ? 'npm'
-    : process.env.PACKAGE_MANAGER_EXECUTABLE.startsWith('yarn')
-    ? 'yarn'
-    : process.env.PACKAGE_MANAGER_EXECUTABLE; // TODO: replace `process.env.PACKAGE_MANAGER_EXECUTABLE` with `getPackageManagerName()` once the test infra is ready.
-
   /**
    * Copies the template directory to an amplify folder within the projectRoot
    */
   generateInitialProjectFiles = async (): Promise<void> => {
-    const targetDir = path.resolve(this.projectRoot, 'amplify');
-    await this.fs.mkdir(targetDir, { recursive: true });
-    await this.fs.cp(
+    const targetDir = this.path.resolve(this.projectRoot, 'amplify');
+    await this.fsp.mkdir(targetDir, { recursive: true });
+    await this.fsp.cp(
       new URL('../templates/basic-auth-data/amplify', import.meta.url),
       targetDir,
       { recursive: true }
     );
 
     const packageJsonContent = { type: 'module' };
-    await this.fs.writeFile(
-      path.resolve(targetDir, 'package.json'),
+    await this.fsp.writeFile(
+      this.path.resolve(targetDir, 'package.json'),
       JSON.stringify(packageJsonContent, null, 2)
     );
 
-    if (process.env.PACKAGE_MANAGER_EXECUTABLE === 'yarn-modern') {
-      fs.writeFile(path.resolve(targetDir, 'yarn.lock'), '', (err) => {
-        if (err) {
-          console.error(`Error creating ${targetDir}/${targetDir}`, err);
-        } else {
-          console.log(`${targetDir}/yarn.lock created successfully.`);
-        }
-      });
+    if (this.packageManagerController.addLockFile) {
+      this.packageManagerController.addLockFile(targetDir);
     }
 
-    await this.initializeTsConfig(targetDir);
-  };
-
-  private initializeTsConfig = async (targetDir: string): Promise<void> => {
-    const tscArgs = [
-      'tsc',
-      '--init',
-      '--resolveJsonModule',
-      'true',
-      '--module',
-      'es2022',
-      '--moduleResolution',
-      'bundler',
-      '--target',
-      'es2022',
-    ];
-
-    if (this.executableName.startsWith('yarn')) {
-      await this.executeWithDebugLogger(
-        targetDir,
-        'yarn',
-        ['add', 'typescript@^5'],
-        execa
-      );
+    if (this.packageManagerController.addTypescript) {
```

Comment:
same as above, this should always call `addTypescript`

Reply:
Done